tests/test_main.py

```diff
@@ -142,3 +142,15 @@ def test_sort_courses(mocker, test_instructor):
     
     mock_sort_courses.assert_called_once_with('a')
 
+
+def test_grades_to_edit(monkeypatch, capsys, test_instructor):
+    # Act & Arrange
+    responses = iter([test_instructor["id"], test_instructor["courses"][0], '1', '201', '99', '', '2', '201', '88', '', 'x', '', 'q'])
+    monkeypatch.setattr('builtins.input', lambda _: next(responses))
+
+    with pytest.raises(SystemExit) as exitInfo:
+        main()
+
+    # Assert
+    captured = capsys.readouterr()
+    assert "Alice (201): 99.0" in captured.out
```

Comment:
Need to add tests for invalid student IDs and invalid grades. Also, need to add tests for courses with no existing grades.

When you test main, you actually just need to make sure edit_grade is called correctly in certain cases.

You also need to create tests for edit_grade function. If there are already tests for edit_grade function, let me know. I just don't have time to check the code to see if they are there.